samcli/lib/sync/flows/alias_version_sync_flow.py

```diff
@@ -61,6 +62,11 @@ def __init__(
         self._function_identifier = function_identifier
         self._alias_name = alias_name
         self._lambda_client = None
+        # Sync state is the unique identifier for each sync flow
+        # In sync state toml file we will store
+        # Key as AliasVersionSyncFlow:FunctionLogicalId:AliasName
+        # Value as alias version number
+        self._sync_state_identifier = self.__class__.__name__ + ":" + self._function_identifier + ":" + self._alias_name
```

Comment:
I wonder if we could do this as a method and calculate this on the fly (to have less fields and more straightforward constructor)? 🤔

Reply:
We can make this a property

Reply:
You mean function with `@property` annotation on top of it? If so I like that idea!

Reply:
yep!